File: src/main/java/io/citytrees/repository/FileRepository.java
package io.citytrees.repository;

import io.citytrees.model.CtFile;
import org.springframework.data.jdbc.repository.query.Query;
import org.springframework.data.repository.CrudRepository;

import java.util.Optional;
import java.util.UUID;

public interface FileRepository extends CrudRepository<CtFile, UUID> {
    @Query("""
        INSERT INTO ct_file(id, name, mime_type, size, hash, user_id)
        VALUES (:id, :name, :mimeType, :size, :hash, :userId)
        RETURNING id
        """)
    UUID save(UUID id, String name, String mimeType, Long size, String hash, UUID userId);

    Optional<CtFile> findFirstByHash(String hash);

    long countByHash(String hash);
}

File: src/main/java/io/citytrees/repository/FileRepository.kt
package io.citytrees.repository

import io.citytrees.constants.TableNames.FILE_TABLE
import io.citytrees.model.CtFile
import org.springframework.data.jdbc.repository.query.Query
import org.springframework.data.repository.CrudRepository
import java.time.LocalDateTime
import java.util.*

interface FileRepository : CrudRepository<CtFile, UUID> {
    @Query(
        """
        INSERT INTO $FILE_TABLE(id, name, mime_type, size, hash, creation_date_time, user_id)
        VALUES (:id, :name, :mimeType, :size, :hash, :creationDateTime, :userId)
        RETURNING id
        """
    )
    fun save(id: UUID, name: String, mimeType: String, size: Long, hash: String, creationDateTime: LocalDateTime, userId: UUID): UUID

    fun findFirstByHash(hash: String): Optional<CtFile>

    fun countByHash(hash: String): Long

    fun findAllByIdIn(uuids: Set<UUID>): List<CtFile>
}